src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4567,7 +4567,16 @@ private ParameterSyntax ParseParameter()
                 }
 
                 var type = this.ParseType(mode: ParseTypeMode.Parameter);
-                var identifier = this.ParseIdentifierToken();
+                SyntaxToken identifier;
+
+                if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken && IsCurrentTokenWhereOfConstraintClause())
```

Comment:
>&& IsCurrentTokenWhereOfConstraintClause() [](http://example.com/codeflow?start=73&length=42)

Is this an error recovery scenario that is specific to `class` and `struct` or could we get here with `record` too? Are we testing the `record` case?

Reply:
>Is this an error recovery scenario that is specific to class and struct or could we get here with record too? 

This is recovery is not specific to any particular type declaration.

>Are we testing the record case?

No. Do you think we have to?